cmd/config.go

```diff
@@ -120,17 +116,45 @@ func setNodeCmdRun(cmd *cobra.Command, args []string) error {
 	}
 }
 
-func importCmdRun(cmd *cobra.Command, args []string) error {
-	vms, err := utils.VirtualMachineScaleSetVMsViaKubeconfig()
-	if err != nil {
-		return fmt.Errorf("failed to get VMSS VMs: %w", err)
+func importCmdCommand() *cobra.Command {
+	var subscriptionID string
+	var resourceGroup string
+	var clusterName string
+
+	cmd := &cobra.Command{
+		Use:          "import",
+		Short:        "Import Kubernetes nodes in the configuration",
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			utils.DefaultSpinner.Start()
+			defer utils.DefaultSpinner.Stop()
+			vms, err := utils.VirtualMachineScaleSetVMsViaKubeconfig()
+			if err != nil {
+				logrus.Warnf("Could not get VMSS VMs via Kubernetes API: %v", err)
+				logrus.Warnf("Trying to get VMSS VMs via Azure API")
+				if subscriptionID == "" || resourceGroup == "" || clusterName == "" {
+					return fmt.Errorf("specify '--subscription-id', '--resource-group' and '--cluster-name' " +
+						"to get VMSS VMs via Azure API")
+				}
+				vms, err = utils.VirtualMachineScaleSetVMsViaAzureAPI(subscriptionID, resourceGroup, clusterName)
+				if err != nil {
+					return fmt.Errorf("getting VMSS VMs via Azure API: %w", err)
+				}
+			}
+
+			cfg := config.New()
+			for nn, vm := range vms {
+				if err = cfg.SetNodeConfigWithVMSSInfoFlag(nn, vm.SubscriptionID, vm.NodeResourceGroup, vm.VMScaleSet, vm.InstanceID); err != nil {
+					return fmt.Errorf("setting node config for %s: %w", nn, err)
+				}
+			}
+			return nil
+		},
 	}
 
-	cfg := config.New()
-	for nn, vm := range vms {
-		if err = cfg.SetNodeConfigWithVMSSInfoFlag(nn, vm.SubscriptionID, vm.NodeResourceGroup, vm.VMScaleSet, vm.InstanceID); err != nil {
-			return fmt.Errorf("failed to set node config for %s: %w", nn, err)
-		}
-	}
-	return nil
+	cmd.Flags().StringVarP(&subscriptionID, "subscription-id", "s", "", "Subscription ID of the cluster")
+	cmd.Flags().StringVarP(&resourceGroup, "resource-group", "g", "", "Resource group of the cluster")
+	cmd.Flags().StringVarP(&clusterName, "cluster-name", "c", "", "Name of the cluster")
```

Comment:
These flags are only needed if k8s mechanism is not used. We should add it to the description, or not if we decide to remove the k8s mechanism. See above comment.